internal/pkg/agent/application/enroll/enroll.go

```diff
@@ -57,6 +57,7 @@ func EnrollWithBackoff(
 	options EnrollOptions,
 	configStore saver,
 	backoffFactory func(done <-chan struct{}) backoff.Backoff,
+	maxRetries int,
```

Comment:
Prefer `maxAttempts` as retries could be misinterpreted as always making one attempt first and then retrying N times.